### DISCUSSION TITLE
Saving future/pending entries to field in custom plugin

### DISCUSSION THREAD

**Discussion by jordancraftltd** [craftcms/cms#9831] [Q&A] [answered]
I am having a few issues with updating an entries field with pending (entries with a future post date) within a custom plugin. I'm not sure if the issue is in the setFieldValue or saveElement function.

I have had a good look online, in the docs and asked the community without any luck so far.

I was also trying to look for a way to simply add one more entry to a field without having to re-add any existing entries assigned to the field. Is this possible?

Condensed code below for a short reference:

```php
$videos = $subscription->subVideos->ids();

$video = Entry::findOne(['section' => 'videos', 'relatedTo' => $channel->id, 'after' => $dateTomorrow, 'status' => 'enabled', 'orderBy' => 'postDate asc']);

if (!in_array($video->id, $videos)) {
    array_push($videos, $video->id);
}

$subscription->setFieldValue([
    'subVideos' => $videos
]);

$saveSubscription = Craft::$app->elements->saveElement($subscription);
```

Any help would be greatly appreciated.

**Comment by brandonkelly** [comment, marked as answer]
Your `setFieldValue()` syntax is wrong. The first argument should be the field handle, and the second argument should be the array of IDs:

```php
$subscription->setFieldValue('subVideos', $videos);
```

You would only pass an array of field handle/value mappings if you were calling `setFieldValues()` (plural) instead:

```php
$subscription->setFieldValues([
    'subVideos' => $videos,
    // ...
]);
```

**Reply by jordancraftltd** [reply]
Thanks Brandon, the function now works as expected :)
